vizro-core/src/vizro/actions/_parameter_action.py

```diff
@@ -24,9 +26,28 @@ def _parameter(targets: List[str], **inputs: Dict[str, Any]) -> Dict[str, Any]:
     """
     target_ids: List[ModelID] = [target.split(".")[0] for target in targets]  # type: ignore[misc]
 
-    return _get_modified_page_figures(
+    figures_outputs = _get_modified_page_figures(
         targets=target_ids,
         ctds_filter=ctx.args_grouping["external"]["filters"],
         ctds_filter_interaction=ctx.args_grouping["external"]["filter_interaction"],
         ctds_parameters=ctx.args_grouping["external"]["parameters"],
     )
+
+    for target in targets:
+        target_id, targeted_argument = target.split(".", 1)
+        if targeted_argument.startswith("data_frame"):
+            from vizro.models._controls import Filter
+            for filter in model_manager._items_with_type(Filter):
+                filter = filter[1]
+                if target_id in filter.targets:
+                    filter._set_categorical_selectors_options(
+                        force_update=True,
+                        data_load_kwargs={
+                            target_id: {
+                                "points": ctx.args_grouping["external"]["parameters"][0]["value"]
+                            }
+                        }
+                    )
+                    figures_outputs[filter.id] = filter.build()
```

Comment:
Will currently be nesting infinitely?

Reply:
Need general "rebuild" action that calls `pre_build` again and then replaces whole `build`?